onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1031,7 +1017,29 @@ Applies to session load, initialization, etc. Default is 0.)pbdoc")
           },
           R"pbdoc(Graph optimization level for this session.)pbdoc")
       .def_readwrite("use_deterministic_compute", &SessionOptions::use_deterministic_compute,
-                     R"pbdoc(Whether to use deterministic compute. Default is false.)pbdoc");
+                     R"pbdoc(Whether to use deterministic compute. Default is false.)pbdoc")
+      .def(
+          "add_free_dimension_override",
```

Comment:
We can call this add_free_dimension_override_by_denotation since that's what the C API does. It was a poor naming choice in the C API.

Reply:
The C API has this - `ORT_API_STATUS_IMPL(OrtApis::AddFreeDimensionOverride ...`. It is missing `ByDenotation`. Do we still want to add it for the Python API ?